File: js/selection_sort.js
// simplest sorting algorithm
// algoritmo mais simples de ordenação

getMin = array => {
    minor = array[0]

    array.forEach(element => {
        if (element < minor) {
            minor = element
        }    
    });

    return minor
}

array1 = [3, 5, 8, 9, 2, 5, 0, 4, 5]
sortedArray = []
size = array1.length
count = 0
console.log('Normal -> ' + array1)

while (count < size) {
    minor = getMin(array1)
    sortedArray.push(minor)
    index = array1.indexOf(minor)
    array1.splice(index, 1)
    count++
}
 
console.log('Sorted -> ' + sortedArray)

// the concept is to get the minor value and to move it to another array,
// then you remove this value from the original array

// a ideia é encontrar o menor valor do array e movê-lo para outro array,
// depois o valor movido é removido do array original

File: py/selection_sort.py
# simplest sorting algorithm
# algoritmo mais simples de ordenação

def getMin(array):
    minor = array[0]
    
    for n in array:
        if n < minor:
            minor = n
    
    return minor

array1 = [3, 5, 8, 9, 2, 5, 0, 4, 5]
sortedArray = []
print('\n')
print('Normal -> ', array1)

count = 0
size = len(array1)

while count < size:
    minor = getMin(array1)
    sortedArray.append(minor)
    array1.remove(minor)
    count += 1
    
print('Sorted -> ', sortedArray)

# the concept is to get the minor value and to move it to another array
# then you remove this value from the original array

# a ideia é encontrar o menor valor do array e movê-lo para outro array
# depois o valor movido é removido do array original